hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/CalciteSqlOptimizer.java

```diff
@@ -291,19 +292,21 @@ public PlanExecutor getPlanExecutor() {
     public SqlPlan prepare(OptimizationTask task) {
         // 1. Prepare context.
         int memberCount = nodeEngine.getClusterService().getSize(MemberSelectors.DATA_MEMBER_SELECTOR);
+        SqlSecurityContext ssc = task.getSqlSecurityContext();
 
         OptimizerContext context = OptimizerContext.create(
                 task.getSchema(),
                 task.getSearchPaths(),
                 task.getArguments(),
                 memberCount,
-                iMapResolver);
+                iMapResolver,
+                ssc);
 
         try {
             OptimizerContext.setThreadContext(context);
 
             // 2. Parse SQL string and validate it.
-            QueryParseResult parseResult = context.parse(task.getSql());
+            QueryParseResult parseResult = context.parse(task.getSql(), ssc);
```

Comment:
`context` already contains `ssc`, no need to pass it again

Reply:
ok, might contain, but you pass it in `OptimizerContext.create` so maybe should store it?

Reply:
https://github.com/hazelcast/hazelcast/pull/25784/commits/898812b5a218576dfbcafa5b86009cd4c3d53c70